### DISCUSSION TITLE
Could not find a field for name x

### DISCUSSION THREAD

**Discussion by elect86** [pdvrieze/xmlutil#112] [Q&A] [answered]
Sorry for the newbie question, I need to parse an xml into a (data) class, which it starts as:

```
<?xml version="1.0" encoding="UTF-8"?>
<registry>
    <comment>
Copyright 2015-2022 The Khronos Group Inc.

SPDX-License-Identifier: Apache-2.0 OR MIT
    </comment>
 ```
 
 This is my code: 

```
fun parse(text: String) {
    val format = XML {}
    val registry = format.decodeFromString<Registry>(text)
}

@Serializable
data class Registry(
    val comment: String,)
 ```

but I get:
> Execution failed for task ':generateCode'.
> Could not find a field for name (vkk.Registry) Registry/comment (Element)
    candidates: Comment at position Line number = 3

What am I missing?

**Comment by pdvrieze** [comment, marked as answer]
When serializing some example data it is clearer. Comment is a string so is by default serialized as attribute, not as element. The error doesn't  clarify this (yet). To fix it you need to make comment serialize as element. The easiest way to do that is the `@XmlElement(true)` annotation. You will also want to set the serial name (or `@XmlSerialName`) for the registry class (as otherwise the capitalised class name is used).
